### PR TITLE
`libraqm`: add missing `fribidi` dependency

### PR DESCRIPTION
**Pull request by thomas-bouvier**

### CODE REVIEW

**Comment by thomas-bouvier** [comment]
@spackbot fix style

**Review by tldahlgren — DISMISSED** [review]
LGTM though I wonder if version `1.0.6` should be added to `fribidi`.

**Comment by tldahlgren** [comment]
Added @certik as a reviewer in case, as the original contributor, you want to comment on this PR.

**Comment by certik** [comment]
I don't remember the context for this unfortunately. The change seems simple?

**Comment by thomas-bouvier** [comment]
The library requires either `fribidi` or `sheenbidi` https://github.com/HOST-Oman/libraqm/blob/master/meson.build#L76-L88

**Comment by tldahlgren** [comment]
> The library requires either `fribidi` or `sheenbidi` https://github.com/HOST-Oman/libraqm/blob/master/meson.build#L76-L88

Do you want to add a variant to choose the option then? (See https://spack.readthedocs.io/en/latest/packaging_guide.html#multi-valued-variants.)

**Comment by thomas-bouvier** [comment]
Will do. I have to add a recipe for `sheenbidi` first.

**Comment by thomas-bouvier** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/libraqm/package.py [hunk 1]
```diff
@@ -15,5 +15,16 @@ class Libraqm(MesonPackage):
...
+        default="fribidi",
+        description="Unicode Bidirectional Algorithm",
+        values=("fribidi", "sheenbidi"),
+        multi=False,
+    )
+
     depends_on("freetype")
     depends_on("harfbuzz")
+
+    depends_on("fribidi@1.0.6:", when="bidi_algo=fribidi")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libraqm/package.py:29]: This version does not appear in the spack package, so not safe.  

It also does not show up in the releases page (https://github.com/fribidi/fribidi/releases) BUT does as a tag at https://github.com/fribidi/fribidi/archive/refs/tags/v1.0.6.tar.gz, which does not have a supported URL format in that package.

File: var/spack/repos/builtin/packages/sheenbidi/package.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class Sheenbidi(MesonPackage):
+    """A sophisticated implementation of Unicode Bidirectional Algorithm"""
+
+    homepage = "https://github.com/Tehreer/SheenBidi"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sheenbidi/package.py:13]: Confirmed the homepage.

File: var/spack/repos/builtin/packages/sheenbidi/package.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    url = "https://github.com/Tehreer/SheenBidi/archive/v2.6.tar.gz"
+    git = "https://github.com/Tehreer/SheenBidi.git"
+
+    version("2.6", sha256="f538f51a7861dd95fb9e3f4ad885f39204b5c670867019b5adb7c4b410c8e0d9")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sheenbidi/package.py:17]: Confirmed the sha256.

**Review by tldahlgren — APPROVED** [review]
The changes LGTM, I can build `sheenbidi` separately, and `spack spec` succeeds for `libraqm`; however, I cannot build `libraqm` with either option because `pixman` (`harfbuzz` dependency) fails to build for me:

```
==> Installing pixman-0.42.2-weaiekpkfogzshtuqvveicf43uzxnong
==> No binary for pixman-0.42.2-weaiekpkfogzshtuqvveicf43uzxnong found: installing from source
==> Using cached archive: /usr/WS1/dahlgren/releases/spack/var/spack/cache/_source-cache/archive/ea/ea1480efada2fd948bc75366f7c349e1c96d3297d09a3fe62626e38e234a625e.tar.gz
==> No patches needed for pixman
==> pixman: Executing phase: 'autoreconf'
==> pixman: Executing phase: 'configure'
==> pixman: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

34 errors found in build log:
...
  >> 665    /tmp/dahlgren/spack-stage/spack-stage-pixman-0.42.2-weaiekpkfogzsht
            uqvveicf43uzxnong/spack-src/test/utils.c:690: undefined reference t
            o `png_init_io'
     666    ./.libs/libutils.a(utils.o): in function `write_png':
...
```

I will approve this PR since this seems to be an issue outside the scope of it.

**Comment by tldahlgren** [comment]
There's a pod issue with one of the CI stacks.

@spackbot run pipeline

**Pull request merged**